benchmarks/graphics_pipeline/main.cpp

```diff
@@ -189,6 +204,124 @@ void ProjApp::Setup()
         mCamera.SetPerspective(60.f, GetWindowAspect());
     }
 
+    // Uniform buffers
+    {
+        grfx::BufferCreateInfo createInfo        = {};
+        createInfo.size                          = 256;
+        createInfo.usageFlags.bits.uniformBuffer = true;
+        createInfo.memoryUsage                   = grfx::MEMORY_USAGE_CPU_TO_GPU;
+
+        PPX_CHECKED_CALL(GetDevice()->CreateBuffer(&createInfo, &mUniformBuffer));
+    }
+
+    // Texture image, view, and sampler
+    {
+        // SkyBox
+        {
+            grfx_util::ImageOptions options = grfx_util::ImageOptions().MipLevelCount(PPX_REMAINING_MIP_LEVELS);
+            PPX_CHECKED_CALL(grfx_util::CreateImageFromFile(GetDevice()->GetGraphicsQueue(), GetAssetPath("basic/models/spheres/basic-skybox.jpg"), &mImage, options, true));
+
+            grfx::SampledImageViewCreateInfo viewCreateInfo = grfx::SampledImageViewCreateInfo::GuessFromImage(mImage);
+            PPX_CHECKED_CALL(GetDevice()->CreateSampledImageView(&viewCreateInfo, &mSampledImageView));
+        }
+
+        // Sampler
+        grfx::SamplerCreateInfo samplerCreateInfo = {};
+        samplerCreateInfo.magFilter               = grfx::FILTER_LINEAR;
+        samplerCreateInfo.minFilter               = grfx::FILTER_LINEAR;
+        samplerCreateInfo.mipmapMode              = grfx::SAMPLER_MIPMAP_MODE_LINEAR;
+        samplerCreateInfo.minLod                  = 0;
+        samplerCreateInfo.maxLod                  = FLT_MAX;
+        PPX_CHECKED_CALL(GetDevice()->CreateSampler(&samplerCreateInfo, &mSampler));
+    }
+
+    // Descriptor
+    {
+        grfx::DescriptorPoolCreateInfo poolCreateInfo = {};
+        poolCreateInfo.uniformBuffer                  = 1024;
+        poolCreateInfo.sampledImage                   = 1024;
+        poolCreateInfo.sampler                        = 1024;
```

Comment:
1024 because you would need them for future spheres?

Reply:
Yes, I just put 1024 to Create descriptor pool large enough for this project, however, with the new pattern, I think I will only need to use uniform buffers to generate up 3000 spheres, so I updated this to 8. Thanks for pointing it.